lib/Analysis/Utility.cpp

```diff
@@ -30,6 +30,9 @@ SmallVector<unsigned> getParentOrder(Attribute layout) {
 } // namespace
 
 bool ReduceOpHelper::isFastReduction() {
+  // We don't need shared memory reduction if everything is within a single
+  // warp
+  if (getIntraWarpSize() == 1) return true;
```

Comment:
IntraWarpSize is the number of threads within a warp that need to be reduced. This doesn't match the comment as it sounds like you want to check `getInterWarpSize()` to know how many warps we need to reduce across.

That being said I don't think the check is correct anyway as `matchAndRewriteFast` makes assumption that the reduce dimension is the fastest moving one therefore we cannot use this path even if all the reduction is done within one warp.

Reply:
Yes, that was a typo. Should have been `Inter`.

Thanks for the explanation about the implicit assumption `matchAndRewriteFast` makes. Then the fix I had in mind doesn't work.

I went digging a bit deeper into this, and figured out that the underlying issue that caused the crash was actually
because we used boolean (`i1`) in the reduction. The utility to calculate the required shared memory size would just
return 0 for sub-byte types.

I updated the PR code and description to fix the underlying issue instead. Could you take a look? Thanks!